policies/dx/RestartDexOnCredentialsRenewal.yaml

```
@@ -0,0 +1,116 @@
+      respective secrets are updated. The dex-operator deployment restarts when
+      dex-operator-credentials changes (credential rotation). The dex-app deployment
+      restarts when its configuration secret changes (after operator updates it).
+spec:
+  background: false
+  schemaValidation: false
+  rules:
+    - name: restart-dex-operator-on-credentials-change
+      match:
+        any:
+        - resources:
+            kinds:
+            - Secret
+            names:
+            - dex-operator-credentials
+            namespaces:
+            - giantswarm
+            - org-*
+            operations:
+            - UPDATE
+      preconditions:
+        all:
+        - key: "[[request.object.data]]"
+          operator: NotEquals
+          value: "[[request.oldObject.data]]"
+      mutate:
+        targets:
+          - apiVersion: apps/v1
+            kind: Deployment
+            name: dex-operator
+            namespace: "[[request.object.metadata.namespace]]"
+        patchStrategicMerge:
+          spec:
+            template:
+              metadata:
+                annotations:
+                  dex-operator.giantswarm.io/restarted-at: "[[time_now()]]"
+                  dex-operator.giantswarm.io/credentials-version: "[[request.object.metadata.resourceVersion]]"
+                  dex-operator.giantswarm.io/restart-reason: "credentials-rotation"
+
+    - name: restart-dex-app-on-config-change
+      match:
+        any:
+        - resources:
+            kinds:
+            - Secret
+            names:
+            - "*-default-dex-config"
+            namespaces:
+            - giantswarm
+            - org-*
+            operations:
+            - UPDATE
+      preconditions:
+        all:
+        - key: "[[request.object.data]]"
+          operator: NotEquals
+          value: "[[request.oldObject.data]]"
+        - key: "[[request.object.metadata.labels.\"giantswarm.io/managed-by\" || '']]"
+          operator: Equals
+          value: "dex-operator"
+      mutate:
+        targets:
+          # Target dex-app deployment based on the secret name pattern
+          # The app name is derived from the secret name by removing "-default-dex-config"
+          - apiVersion: apps/v1
+            kind: Deployment
+            name: dex-app
+            namespace: "[[request.object.metadata.namespace]]"
+        patchStrategicMerge:
+          spec:
+            template:
+              metadata:
+                annotations:
+                  dex-operator.giantswarm.io/restarted-at: "[[time_now()]]"
+                  dex-operator.giantswarm.io/config-version: "[[request.object.metadata.resourceVersion]]"
+                  dex-operator.giantswarm.io/restart-reason: "config-update"
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels:
+    application.giantswarm.io/team: honeybadger
+    rbac.kyverno.io/aggregate-to-admission-controller: "true"
```

```suggestion
    rbac.kyverno.io/aggregate-to-background-controller: "true"
```

Actually this would be background-controller since it's a mutation